client/verta/verta/_deployment/endpoint.py

```diff
@@ -76,9 +77,66 @@ def _get_json_by_path(cls, conn, workspace, path):
         # TODO: iterate through response.json().get('endpoints', []) to find matching creator_request.path
 
     def update(self, run, strategy):
-        raise NotImplementedError
-        # TODO: check if isinstance(run, experimentrun.ExperimentRun)
-        # TODO: check if isinstance(strategy, deployment._UpdateStrategy)
+        if not isinstance(run, experimentrun.ExperimentRun):
+            raise TypeError("run must be an ExperimentRun")
+
+        if not issubclass(strategy, deployment._UpdateStrategy):
+            raise TypeError("strategy must be an _UpdateStrategy")
```

Comment:
My suggested wording is still a little awkward, but a user shouldn't have to be concerned about what an `_UpdateStrategy` is.
```suggestion
            raise TypeError("strategy must be an object from verta.deployment.strategies")
```

Reply:
Fixed.